ncm-metaconfig/src/main/metaconfig/rsyslog/pan/actions.pan

```diff
@@ -0,0 +1,143 @@
+declaration template metaconfig/rsyslog/actions;
+
+@{Action/output module schema configuration options}
+
+type rsyslog_action_options = {
+    'writeAllMarkMessages' ? boolean
+    'execOnlyEveryNthTime' ? long(0..)
+    'execOnlyEveryNthTimeout' ? long(0..)
+    'execOnlyOnceEveryInterval' ? long(0..)
+    'execOnlyWhenPreviousIsSuspended' ? long(0..)
+    'repeatedmsgcontainsoriginalmsg' ? boolean
+    'resumeRetryCount' ? long(-1..)
+    'resumeInterval' ? long(0..)
+    'reportSuspension' ? boolean
+    'reportSuspensionContinuation' ? boolean
+    'copyMsg' ? boolean
+};
+
+type rsyslog_action_common = {
+    @{name (useful for debugging)}
+    'name' ? string
+    @{action options}
+    'options' ? rsyslog_action_options
+};
+
+@{output file common module and action parameters}
+type rsyslog_action_file_common_module = {
+    'template' ? string
+    'dirCreateMode' ? long(0..)
+    'fileCreateMode' ? long(0..)
+    'fileOwner' ? defined_user
+    'fileOwnerNum' ? long(0..)
+    'fileGroup' ? defined_group
+    'fileGroupNum' ? long(0..)
+    'dirOwner' ? defined_user
+    'dirOwnerNum' ? long(0..)
+    'dirGroup' ? defined_group
+    'dirGroupNum' ? long(0..)
+};
+
+type rsyslog_action_file = {
+    include rsyslog_action_common
+    include rsyslog_action_file_common_module
+    'file' ? string
+    'dynaFile' ? string
+    'closeTimeout' ? long(0..)
+    'dynaFileCacheSize' ? long(0..)
+    'zipLevel' ? long(0..)
+    'veryRobustZip' ? boolean
+    'flushInterval' ? long(0..)
+    'asyncWriting' ? boolean
+    'flushOnTXEnd' ? boolean
+    'ioBufferSize' ? long(0..)
+    'failOnChOwnFailure' ? boolean
+    'createDirs' ? boolean
+    'sync' ? boolean
+    'sig.provider' ? string with match(SELF, '^(ksi|gt|ksi_ls12)$')
+    'cry.provider' ? string with match(SELF, '^(gcry)$')
+};
+
+type rsyslog_action_prog = {
+    include rsyslog_action_common
+    @{The binary (and command line options; make sure to esacpe the double quotes)}
+    'binary' : string
+    'hup.signal' ? string with match(SELF, '^(HUP|USR1|USR2|INT|TERM)$')
+    'signalOnClose' ? boolean
+};
+
+type rsyslog_action_fwd = {
+    include rsyslog_action_common
+    'Target' : type_hostname
+    'Port' ? long(0..)
+    'Protocol' ? string with match(SELF, '^(udp|tcp)$')
+    'NetworkNamespace' ? string
+    'Device' ? string with exists("/hardware/cards/nic/" + SELF)
+    'TCP_Framing' ? string with match(SELF, '^(traditional|octet-counted)$')
+    'ZipLevel' ? long(0..9)
+    'maxErrorMessages' ? long(0..)
+    'compression.mode' ? string with match(SELF, '^(none|single|stream:always)$')
+    'compression.stream.flushOnTXEnd' ? boolean
+    'RebindInterval' ? long(0..)
+    'KeepAlive' ? boolean
+    'KeepAlive.Probes' ? long(0..)
+    'KeepAlive.Interval' ? long(0..)
+    'KeepAlive.Time' ? long(0..)
+    'StreamDriver' ? string
+    'StreamDriverMode' ? long(0..)
+    'StreamDriverAuthMode' ? string
```

Comment:
I guess this would be a good use for the `choice` type.

>   anon - anonymous authentication as described in IETF's draft-ietf-syslog-transport-tls-12 Internet draft
    x509/fingerprint - certificate fingerprint authentication as described in IETF's draft-ietf-syslog-transport-tls-12 Internet draft
    x509/certvalid - certificate validation only
    x509/name - certificate validation and subject name authentication as described in IETF's draft-ietf-syslog-transport-tls-12 Internet draft